engine/runtime/src/main/scala/org/enso/compiler/context/ChangesetBuilder.scala

```diff
@@ -42,8 +43,43 @@ final class ChangesetBuilder[A: TextEditor: IndexedSource](
     * @return the computed changeset
     */
   @throws[CompilerError]
-  def build(edits: Seq[TextEdit]): Changeset[A] =
-    Changeset(source, ir, compute(edits))
+  def build(edits: Seq[TextEdit]): Changeset[A] = {
+
+    def findSimpleChange(): (IR.Literal, TextEdit) = {
+      val edit = if (edits.size != 1) {
+        if (edits.size != 2) {
+          return null
+        }
+        try {
+          val firstAffected = invalidated(Seq(edits.head))
```

Comment:
The IDE sends two edits, usually even for _simple edit_. The other edit is modifying a comment with node IDs at the end of the file. What would be the best way to ignore such edit, @kustosz , @4e6 ? Are the IDs treated especially in the IR? Should a comment only edit be ignore? Should we know _last executable line_ in the code and ignore anything after it? Thank you in advance for your insights.

Reply:
There is no reliable way to say if the edit changes to the code or metadata. But IDE knows what it edits, so it can tell in the `text/applyEdit` command if the re-execution needed or not.

Reply:
Expanding the protocol is an option. Just it is not about _re-execution_, but about re-parsing whole source. Right now we can handle _simple edits_ that replace number and text literals. If the IDE can label such edits somehow in advance, the engine will not have to use _heuristic_ to detect it.